### PR TITLE
Ruby 3.1.0

### PR DESCRIPTION
**Pull request by ramonskie**
this will bump the bosh director to use ruby 3.1
this is needed as ruby 3.1 supports openssl 3
which is now the default library used in `jammy(ubuntu 22.04)`

my only concern so far is that we will get a cascading effect by bumping the docker containers to use a new ruby version.
if the use the latest tag.

### CODE REVIEW

**Review by beyhan — CHANGES_REQUESTED** [review]
@ramonskie @bgandon why there is a `[WIP] bump ruby to 3.1.0` commit? Is this pr not ready for review yet?

**Comment by ramonskie** [comment]
> @ramonskie @bgandon why there is a `[WIP] bump ruby to 3.1.0` commit? Is this pr not ready for review yet?

well it was the first commit when me and benjamin where pairing on it.
do you want me to squash this commit?

**Comment by beyhan** [comment]
> well it was the first commit when me and benjamin where pairing on it. do you want me to squash this commit?

@ramonskie  please squash it. There are any way merge conflicts and the pr needs to be touched. Otherwise it looks good to me.

File: src/vendor/cache/extensions/x86_64-linux/3.1.0/eventmachine-abe347b824e3/gem_make.out [hunk 1]
```diff
@@ -0,0 +1,16 @@
+current directory: /home/ramonskie/workspace/bosh/bosh/src/vendor/cache/eventmachine-abe347b824e3/ext/fastfilereader
+/home/ramonskie/.rbenv/versions/3.1.0/bin/ruby -I /home/ramonskie/.rbenv/versions/3.1.0/lib/ruby/3.1.0 -r ./siteconf20220114-83401-11p8t.rb extconf.rb
+creating Makefile
+
+current directory: /home/ramonskie/workspace/bosh/bosh/src/vendor/cache/eventmachine-abe347b824e3/ext/fastfilereader
+make DESTDIR\= clean
+
+current directory: /home/ramonskie/workspace/bosh/bosh/src/vendor/cache/eventmachine-abe347b824e3/ext/fastfilereader
```
Comment (rkoster) [inline src/vendor/cache/extensions/x86_64-linux/3.1.0/eventmachine-abe347b824e3/gem_make.out:8]: We should probably ignore the src/vendor/cache/extensions directory. Will create a separate PR for that.

**Review by rkoster — APPROVED** [review]
Just a bunch of unrelated extensions changes, for the rest looks good to me

**Pull request merged**